src/util/privkey.rs

```diff
@@ -139,34 +144,46 @@ impl FromStr for Privkey {
             network: network,
             key: key
         })
+	}
+}
+
+impl fmt::Display for Privkey {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+		self.fmt_wif(fmt)
+    }
+}
+
+impl FromStr for Privkey {
+    type Err = encode::Error;
+    fn from_str(s: &str) -> Result<Privkey, encode::Error> {
+		Privkey::from_wif(s)
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::Privkey;
     use secp256k1::Secp256k1;
-    use std::str::FromStr;
     use network::constants::Network::Testnet;
     use network::constants::Network::Bitcoin;
 
     #[test]
     fn test_key_derivation() {
         // testnet compressed
-        let sk = Privkey::from_str("cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy").unwrap();
+        let sk = Privkey::from_wif("cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy").unwrap();
         assert_eq!(sk.network(), Testnet);
         assert_eq!(sk.is_compressed(), true);
-        assert_eq!(&sk.to_string(), "cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy");
+        assert_eq!(&sk.to_wif(), "cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy");
```

Comment:
Ditto here -- just leave the `to_string` call alone

Reply:
Added string section in the test.